contracts/src/BattleShip.sol

```diff
@@ -15,7 +21,7 @@ contract BattleShip is IGame {
         number++;
     }
 
-    function moveResult(uint256 moveId, uint8 result, uint256 gameId) external {
+    function moveResult(uint256 moveId, uint8 result, uint256 gameId, ZoKratesStructs.Proof calldata proof) external {
```

Comment:
You have omitted the ZoKrates inputs, which are essential for validation alongside the ZoKrates proof.

Reply:
Please note that zokrates inputs are dependent on the implementation, so you cannot use a fixed array size in the interface.